code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -238,13 +238,27 @@ func (zcn *ZCNSmartContract) CollectRewards(
 	return "", nil
 }
 
-func (zcn *ZCNSmartContract) DeleteAuthorizer(tran *transaction.Transaction, _ []byte, ctx cstate.StateContextI) (string, error) {
+func (zcn *ZCNSmartContract) DeleteAuthorizer(tran *transaction.Transaction, input []byte, ctx cstate.StateContextI) (string, error) {
 	var (
 		authorizerID = tran.ClientID
 		errorCode    = "failed to delete authorizer"
 		err          error
 	)
 
+	params := DeleteAuthorizerPayload{}
+	err = params.Decode(input)
+	if err != nil {
+		err = common.NewError(errorCode, "failed to decode DeleteAuthorizerPayload")
+		Logger.Error("payload decoding error", zap.Error(err))
+		return "", err
+	}
+
+	if params.ID != tran.ClientID {
```

Comment:
this transaction should be allowed only for DelegateWallet, so please check tran.ClientID against delegate wallet